utils/db_utils.py

```diff
@@ -0,0 +1,72 @@
+from datetime import datetime, timezone
+
+import pandas as pd
+
+import emission.core.get_database as edb
+
+from utils.permissions import get_trips_columns
+
+
+def query_uuids(start_date, end_date):
+    query = {'update_ts': {'$exists': True}}
+    if start_date is not None:
+        start_time = datetime.combine(start_date, datetime.min.time()).astimezone(timezone.utc)
+        query['update_ts']['$gte'] = start_time
+
+    if end_date is not None:
+        end_time = datetime.combine(end_date, datetime.max.time()).astimezone(timezone.utc)
+        query['update_ts']['$lt'] = end_time
+
+    projection = {
+        '_id': 0,
+        'user_id': '$uuid',
+        'user_token': '$user_email',
+        'update_ts': 1
+    }
+
+    query_result = edb.get_uuid_db().find(query, projection)
+    df = pd.json_normalize(list(query_result))
+    if not df.empty:
+        df['update_ts'] = pd.to_datetime(df['update_ts'])
+        df['user_id'] = df['user_id'].apply(str)
+    return df
+
+def query_confirmed_trips(start_date, end_date):
+    query = {
+        '$and': [
+            {'metadata.key': 'analysis/confirmed_trip'},
+            {'data.start_ts': {'$exists': True}},
+            {'data.user_input.trip_user_input': {'$exists': False}},
+        ]
+    }
+    if start_date is not None:
+        start_time = datetime.combine(start_date, datetime.min.time())
+        query['$and'][1]['data.start_ts']['$gte'] = start_time.timestamp()
+
+    if end_date is not None:
+        end_time = datetime.combine(end_date, datetime.max.time())
+        query['$and'][1]['data.start_ts']['$lt'] = end_time.timestamp()
+
+    projection = {
+        '_id': 0,
+        'user_id': 1,
+        'trip_start_time_str': '$data.start_fmt_time',
+        'trip_end_time_str': '$data.end_fmt_time',
+        'timezone': '$data.start_local_dt.timezone',
+        'start_coordinates': '$data.start_loc.coordinates',
+        'end_coordinates': '$data.end_loc.coordinates',
+        'travel_modes': '$data.user_input.trip_user_input.data.jsonDocResponse.data.travel_mode',

```

Comment:
so I think this is the list of columns to you want to extract from the `confirmed_trips` "table" (collection in mongodb).
Note that you should really be using the methods from the timeseries here instead of running your own projection code. If the methods you need don't exist, ask for them.

Regardless, either as a projection on the DB itself, or after you have read it into a pandas dataframe, you will need to select the columns that you need to display.

Some of these columns can and should be standard, such as the distance, duration, and so on.
Some of the other columns should be read from the dynamic config. 
So the config should have an entry for "additional_columns" and that will represent the strings mapping into the document for this study or program.

So for UNSW studies, this will be `$data.user_input.trip_user_input.data.jsonDocResponse.data.travel_mode`
For MULTILABEL studies, this will be `$data.user_input.mode_confirm.label`
For some other study, where they don't even care about the mode but only about the amount of range anxiety
`$data.user_input.trip_user_input.jsonDocResponse.data.range_anxiety`

This could look like

```
additionalColumns: {
   {label: 'range_anxiety', 'path': $data.user_input.trip_user_input.jsonDocResponse.data.range_anxiety
}
```

Reply:
solved in https://github.com/e-mission/op-admin-dashboard/pull/7/commits/8a3c37da21c10652f593d1a6d2be8dacb1aa47b2